### PR TITLE
feat: test using cmdtest and fix issues

### PR DESCRIPTION
**Pull request by sermojohn**
This patch adds tests using [cmdtest](https://github.com/google/go-cmdtest) for gitmono commands and validates output and exit code.
Additionally, it fixes a bug with enabling/disabling verbose logs using the `-v` flag.

### CODE REVIEW

File: cmd/gitmono/main.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ type Options struct {
 	Project       string `short:"p" default:"." description:"The project directory to operate on"`
 	Verbose       bool   `short:"v" description:"Enable verbose loggging"`
 	DryRun        bool   `long:"dry-run" description:"Do not persist any write action"`
-	CommitScheme  string `long:"commit-scheme" default:"common" description:"The scheme parse commit messages with (common, conventional)"`
+	CommitScheme  string `long:"commit-scheme" default:"conventional" description:"The scheme parse commit messages with (common, conventional)"`
```
Comment (sermojohn) [inline cmd/gitmono/main.go:22]: This is a breaking change, because we switch commit scheme default to conventional.

File: cmd/gitmono/main.go [hunk 2]
```diff
@@ -58,26 +64,33 @@ func main() {
 
 	// inject options to global component
 	flagsParser.CommandHandler = func(command flags.Commander, args []string) error {
+		// options are ready to use
+		log.SetOutput(ioutil.Discard)
+		if opts.Verbose {
+			log.SetFlags(log.Flags() &^ (log.Ldate | log.Ltime))
```
Comment (sermojohn) [inline cmd/gitmono/main.go:70]: Removing datetime from logs to enable cmdtest for logs. Still considering the idea of logs testing :)

**Review by vkotronis — COMMENTED** [review]
Very interesting use of https://github.com/google/go-cmdtest seems to be the right tool for the job!
No objections on the code fixes/extensions on my side. I will try to provide some additional input on the actual tests by tmr, otherwise please feel free to merge.

**Comment by sermojohn** [comment]
> Very interesting use of https://github.com/google/go-cmdtest seems to be the right tool for the job!
> No objections on the code fixes/extensions on my side. I will try to provide some additional input on the actual tests by tmr, otherwise please feel free to merge.

@vkotronis, thanks a lot for your feedback!

**Pull request merged**